nncf/experimental/onnx/graph/nncf_graph_builder.py

```diff
@@ -0,0 +1,136 @@
+        for output_node in nncf_graph.get_all_nodes():
+            output_node_id = output_node.node_id
+            outputs = onnx_graph.get_all_node_outputs(output_node.node_name)
+            for output in outputs:
+                nodes = onnx_graph.get_nodes_by_input(output)
+                shape = onnx_graph.get_node_output_shape(output)
+                for in_node in nodes:
+                    in_node_id = nncf_graph.get_node_by_name(in_node.name).node_id
+                    input_counter[in_node_id] = input_counter.get(in_node_id, -1) + 1
+                    output_counter[output_node_id] = input_counter.get(output_node_id, -1) + 1
+                    nncf_graph.add_edge_between_nncf_nodes(
+                        from_node_id=output_node_id,
+                        to_node_id=in_node_id,
+                        tensor_shape=shape,
+                        input_port_id=input_counter[in_node_id],
+                        output_port_id=output_counter[output_node_id],
+                        dtype=Dtype.FLOAT
+                    )
+        # Add Input Nodes
+        for i, _input in enumerate(onnx_graph.get_all_model_inputs()):
+            tensor_type = _input.type.tensor_type
+            input_shape = []
+            if tensor_type.HasField("shape"):
+                for d in tensor_type.shape.dim:
+                    if d.HasField("dim_value"):
+                        input_shape.append(int(d.dim_value))
+                    else:
+                        raise RuntimeError('There is no integer value in model Input dimension')
+            else:
+                raise RuntimeError('There is no shape in model Inputs')
+            input_node = nncf_graph.add_nncf_node(node_name='input_node_' + str(i),
+                                                  node_type=NNCFGraphNodeType.INPUT_NODE,
+                                                  node_metatype=ONNX_OPERATION_METATYPES.
+                                                  get_operator_metatype_by_op_name(
+                                                      NNCFGraphNodeType.INPUT_NODE),
+                                                  layer_attributes=None)
+            input_name = _input.name
+            to_nodes = onnx_graph.get_nodes_by_input(input_name)
+            for node in to_nodes:
+                in_node_id = input_node.node_id
+                to_node_id = nncf_graph.get_node_by_name(node.name).node_id
+                input_counter[in_node_id] = input_counter.get(input_node.node_id, -1) + 1
+                output_counter[to_node_id] = input_counter.get(to_node_id, -1) + 1
+                nncf_graph.add_edge_between_nncf_nodes(
+                    from_node_id=input_node.node_id,
+                    to_node_id=to_node_id,
+                    tensor_shape=input_shape,
+                    input_port_id=input_counter[in_node_id],
+                    output_port_id=output_counter[to_node_id],
+                    dtype=Dtype.FLOAT
```

Comment:
What about paths in the graph that are integer, such as picking a dimension value from the result of the `ShapeOf` operation and putting it into a `Gather` operation downstream? Remember Mask-RCNN.

Reply:
Please, take a look at the new version

Reply:
Please extend tests to cover the new logic.

Reply:
I added one model with `Shape `node which output is int